src/tiledbsoma_ml/gpu_batches.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2021-2024 The Chan Zuckerberg Initiative Foundation
+# Copyright (c) 2021-2024 TileDB, Inc.
+#
+# Licensed under the MIT License.
+from __future__ import annotations
+
+import logging
+from typing import Iterable, Iterator
+
+import attrs
+import numpy as np
+import pandas as pd
+from scipy import sparse
+
+from tiledbsoma_ml._utils import EagerIterator
+from tiledbsoma_ml.common import Batch
+from tiledbsoma_ml.io_batches import IOBatches
+
+logger = logging.getLogger("tiledbsoma_ml.gpu_batches")
+
+
+@attrs.define(frozen=True)
+class GPUBatches(Iterable[Batch]):
```

Comment:
I find the naming confusing -- these are not partitioned for a GPU, but are rather what is commonly known as a mini-batch.

Recommend sticking with standard nomenclature from the field

Reply:
On top of that this might be just a personal preference and feel free to omit. I would preferably change the name so it does not hide/shadow the functionality of this class. Since it is an iterator a camel-case `MiniBatchIterator` similar to `EagerIterator` would make it easier traceable and self-explanatory.